pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -2243,6 +2243,176 @@ def is_hermitian(self):
         return True
 
 
+class IntegerComparator(Operation):
+    r"""IntegerComparator(value, geq, control_wires, wires)
+    Apply a Pauli X gate controlled on the following condition:
+
+    Given a basis state :math:`\vert \sigma \rangle`, where :math:`\sigma \in \mathbb{N}`, and a fixed positive
+    integer :math:`L`, flip a target qubit if :math:`\sigma \geq L`. Alternatively, the flipping condition can
+    be :math:`\sigma < L`.
+
+    **Details:**
+
+    * Number of wires: Any (the operation can act on any number of wires)
+    * Number of parameters: 0
+    * Gradient recipe: None
+
+    Args:
+        value (int): The value :math:`L` that the state's decimal representation is compared against.
+        geq (bool): If set to `True`, the comparison made will be :math:`\sigma \geq L`. If `False`, the comparison
+            made will be :math:`\sigma < L`.
+        control_wires (Union[Wires, Sequence[int], or int]): Deprecated way to indicate the control wires.
+            Now users should use "wires" to indicate both the control wires and the target wire.
+        wires (Union[Wires, Sequence[int], or int]): control wire(s) followed by a single target wire where
+            the operation acts on.
+    """
+    is_self_inverse = True
+    num_wires = AnyWires
+    num_params = 0
+    """int: Number of trainable parameters that the operator depends on."""
+
+    grad_method = None
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        value,
+        geq=True,
+        control_wires=None,
+        wires=None,
+        do_queue=True,
+    ):
+
+        if value is None:
+            raise ValueError("Must specify the integer value to compare against.")
```

Comment:
`value` is positional so it cannot be `None`, unless `None` is being passed explicitly. No need to check that - the docstring states that an integer is expected.